src/python.cpp

```diff
@@ -136,7 +136,7 @@ std::string as_std_string(PyObject* str) {
 
   // conver to bytes if its unicode
   PyObjectPtr pStr;
-  if (PyUnicode_Check(str)) {
+  if (PyUnicode_Check(str) || isPyArrayScalar(str)) {
```

Comment:
Presumably at this point we don't need an additional check that the PyArrayScalar is in fact a string scaler? (i.e. the callers of as_std_string have already established that the Python object is a string?)

Reply:
I reviewed the various calls to `as_std_string()` and it does in fact look like we always know that we have a string.